jupyterlab_bigquery/src/components/details_panel/table_details_panel.tsx

```diff
@@ -45,6 +46,14 @@ export default class TableDetailsPanel extends React.Component<Props, State> {
     };
   }
 
+  componentDidMount() {
+    this.mounted = true;
+  }
+
+  componentWillUnmount() {
+    this.mounted = false;
+  }
+
```

Comment:
The `mounted` flag will check whether the tabs are mounted before updating state. This is to ensure we won't try to update an unmounted component and prevents the `Can't perform a React state update on an unmounted component` warning in the console.